backend-transit/src/main/java/eu/socialedge/hermes/backend/transit/domain/GoogleMapsShapeFactory.java

```diff
@@ -40,50 +48,65 @@ public GoogleMapsShapeFactory(String apiKey) {
 
     @Override
     public Shape create(List<Location> waypoints) {
-        val shapePoints = new ArrayList<ShapePoint>(notEmpty(waypoints).size());
-        shapePoints.add(new ShapePoint(waypoints.get(0), Quantities.getQuantity(0, Units.METRE)));
-
-        int chunkNumber = 0;
-        do {
-            val endIndex = chunkNumber + LOCATIONS_LIMIT < waypoints.size() ?
-                chunkNumber + LOCATIONS_LIMIT : waypoints.size();
-            val waypointsChunk = waypoints.subList(chunkNumber, endIndex);
-            val distanceMatrix = calculateDistanceMatrix(waypointsChunk);
-
-            for (int i = 0; i < distanceMatrix.rows.length - 1; i++) {
-                val element = distanceMatrix.rows[i].elements[i + 1];
-                val pointLocation = waypointsChunk.get(i + 1);
-
-                if (element.status != DistanceMatrixElementStatus.OK) {
-                    throw new ShapeFactoryException("Couldn't recognize location " + pointLocation);
-                }
-
-                val localDistance = Quantities.getQuantity(element.distance.inMeters, Units.METRE);
-                val distanceFromOrigin = localDistance.add(shapePoints.get(i).getDistanceTraveled());
-                shapePoints.add(new ShapePoint(pointLocation, distanceFromOrigin));
-            }
-            chunkNumber += LOCATIONS_LIMIT - 1;
-        } while (chunkNumber < waypoints.size() - 2);
+        val googleReadyWaypoints = notEmpty(waypoints).stream()
+            .map(l -> new LatLng(l.getLatitude(), l.getLongitude()))
+            .toArray(LatLng[]::new);
+
+        List<ShapePoint> shapePoints = calculateDistanceTraveled(googleReadyWaypoints).stream()
```

Comment:
unfortunately, lombok cant handle this :(
Error:(61, 26) java: incompatible types: java.util.List<java.lang.Object> cannot be converted to java.util.Collection<eu.socialedge.hermes.backend.transit.domain.ShapePoint>